src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,235 @@
+package mate.jdbc.dao;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.jdbc.lib.Dao;
+import mate.jdbc.lib.exception.DataProcessingException;
+import mate.jdbc.model.Car;
+import mate.jdbc.model.Driver;
+import mate.jdbc.model.Manufacturer;
+import mate.jdbc.util.ConnectionUtil;
+
+@Dao
+public class CarDaoImpl implements CarDao {
+    @Override
+    public Car create(Car car) {
+        String insertRequest = "INSERT INTO cars (title, manufacturer_id) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement insertStatement
+                        = connection.prepareStatement(insertRequest,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            insertStatement.setString(1, car.getTitle());
+            insertStatement.setLong(2, car.getManufacturer().getId());
+            insertStatement.executeUpdate();
+            ResultSet resultSet = insertStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                Long id = resultSet.getObject(1, Long.class);
+                car.setId(id);
+            }
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Couldn't create a car: " + car, throwable);
+        }
+        addDriversToCar(car);
+        return car;
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        String query = "SELECT c.id, c.title, c.manufacturer_id, m.title, m.country "
+                + "FROM cars c JOIN manufacturers m ON c.manufacturer_id = m.id"
+                + " WHERE c.id = ? AND c.is_deleted = false";
+        Car car = new Car();
+        try (Connection connection = ConnectionUtil.getConnection();
+                    PreparedStatement getCarStatement
+                                = connection.prepareStatement(query)) {
+            getCarStatement.setLong(1, id);
+            ResultSet resultSet = getCarStatement.executeQuery();
+            if (resultSet.next()) {
+                car.setId(id);
+                car.setManufacturer(getManufacturerFromResultSet(resultSet));
+                car.setTitle(resultSet.getString(2));
+                car.setDrivers(getDriversById(id, connection));
```

Comment:
you are opening a PreparedStatement inside already opened PreparedStatement - we have already discussed this